packages/sdk/android/src/main/java/com/primerioreactnative/components/manager/redirect/PrimerRNHeadlessUniversalCheckoutComponentWithRedirectManager.kt

```diff
@@ -0,0 +1,250 @@
+package com.primerioreactnative.components.manager.redirect
+
+import android.util.Log
+import androidx.lifecycle.LifecycleOwner
+import androidx.lifecycle.ViewModelStoreOwner
+import androidx.lifecycle.lifecycleScope
+import com.facebook.react.bridge.Arguments
+import com.facebook.react.bridge.Promise
+import com.facebook.react.bridge.ReactApplicationContext
+import com.facebook.react.bridge.ReactContextBaseJavaModule
+import com.facebook.react.bridge.ReactMethod
+import com.facebook.react.bridge.WritableMap
+import com.facebook.react.modules.core.DeviceEventManagerModule
+import com.primerioreactnative.PrimerRNViewModelStoreOwner
+import com.primerioreactnative.components.events.PrimerHeadlessUniversalCheckoutRedirectManagerEvent
+import com.primerioreactnative.components.manager.raw.PrimerRNHeadlessUniversalCheckoutRawManager
+import com.primerioreactnative.datamodels.ErrorTypeRN
+import com.primerioreactnative.datamodels.PrimerInputValidationErrorRN
+import com.primerioreactnative.extensions.toPrimerIssuingBankRN
+import com.primerioreactnative.utils.convertJsonToMap
+import com.primerioreactnative.utils.errorTo
+import io.primer.android.components.SdkUninitializedException
+import io.primer.android.components.domain.exception.UnsupportedPaymentMethodManagerException
+import io.primer.android.components.manager.banks.composable.BanksCollectableData
+import io.primer.android.components.manager.banks.composable.BanksStep
+import io.primer.android.components.manager.componentWithRedirect.PrimerHeadlessUniversalCheckoutComponentWithRedirectManager
+import io.primer.android.components.manager.componentWithRedirect.component.BanksComponent
+import io.primer.android.components.manager.core.composable.PrimerValidationStatus
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.flow.collectLatest
+import kotlinx.coroutines.launch
+import kotlinx.serialization.encodeToString
+import kotlinx.serialization.json.Json
+import org.json.JSONArray
+import org.json.JSONObject
+import kotlinx.coroutines.coroutineScope
+
+class PrimerRNHeadlessUniversalCheckoutComponentWithRedirectManager(
+  private val reactContext: ReactApplicationContext,
+) : ReactContextBaseJavaModule(reactContext) {
+  override fun getName() = "RNTPrimerHeadlessUniversalCheckoutComponentWithRedirectManager"
+
+  private var job: Job? = null
+  private var banksComponent: BanksComponent? = null
+
+  @ReactMethod
+  fun configure(paymentMethodType: String, promise: Promise) {
+
+    val viewModelStoreOwner = reactContext.currentActivity as? ViewModelStoreOwner ?: run {
+      PrimerRNViewModelStoreOwner()
+    }
+
+    banksComponent =
+      PrimerHeadlessUniversalCheckoutComponentWithRedirectManager(viewModelStoreOwner).provide<BanksComponent>(
+        paymentMethodType = paymentMethodType
+      )
+
+    val lifecycleOwner = reactContext.currentActivity as? LifecycleOwner ?: run {
```

Comment:
[q]: should we fallback to custom `CoroutineScope(SupervisorJob() + Dispatchers.Main)`?